src/test/kotlin/it/pagopa/ecommerce/eventdispatcher/queues/v2/TransactionExpirationQueueConsumerTests.kt

```diff
@@ -3258,4 +3282,333 @@ class TransactionExpirationQueueConsumerTests {
               errorCategory =
                 DeadLetterTracedQueueAsyncClient.ErrorCategory.REFUND_MANUAL_CHECK_REQUIRED)))
     }
+
+  @Nested
+  inner class StuckAuthorizationRequestedTransactionTest {
+    @Test
+    fun `messageReceiver should perform a refund from AUTHORIZATION_REQUEST status`() {
+      val events =
+        listOf(
+          transactionActivateEvent(
+            NpgTransactionGatewayActivationData(
+              UUID.randomUUID().toString(), UUID.randomUUID().toString())),
+          transactionAuthorizationRequestedEvent(
+            TransactionAuthorizationRequestData.PaymentGateway.NPG,
+            NpgTransactionGatewayAuthorizationRequestedData(
+              URI.create("http://localhost"),
+              "MC",
+              UUID.randomUUID().toString(),
+              UUID.randomUUID().toString())))
+      given(checkpointer.success()).willReturn(Mono.empty())
+      setupTransactionStorageMock(events)
+      given { authorizationStateRetrieverService.getStateNpg(any(), any(), any(), any(), any()) }
+        .willAnswer {
+          StateResponseDto()
+            .state(WorkflowStateDto.PAYMENT_COMPLETE)
+            .operation(
+              OperationDto()
+                .operationId(UUID.randomUUID().toString())
+                .orderId(UUID.randomUUID().toString())
+                .operationResult(OperationResultDto.EXECUTED)
+                .paymentEndToEndId(UUID.randomUUID().toString())
+                .operationTime(ZonedDateTime.now().toString()))
+            .toMono()
+        }
+
+      given { refundService.requestNpgRefund(any(), any(), any(), any(), any(), any()) }
+        .willAnswer {
+          RefundResponseDto()
+            .operationId(it.arguments[0].toString())
+            .operationTime(ZonedDateTime.now().toString())
+            .toMono()
+        }
+      given(
+          deadLetterTracedQueueAsyncClient.sendAndTraceDeadLetterQueueEvent(
+            capture(binaryDataCaptor), any()))
+        .willReturn(mono {})
+
+      transactionExpirationQueueConsumer
+        .messageReceiver(
+          Either.left(
+            QueueEvent(
+              events.filterIsInstance<TransactionActivatedEvent>().first(), MOCK_TRACING_INFO)),
+          checkpointer,
+          MessageHeaders(mapOf()),
+        )
+        .block()
+
+      verify(transactionsRefundedEventStoreRepository, times(2)).save(any())
```

Comment:
add assertion against transaction-view being saved twice with the expected statuses (REFUND_REQUESTED and REFUNDED)

Reply:
Solved by https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/pull/255/commits/a03dde092c41a59ff7dcefdfb6f0a5b0a4585274